types/beats/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// Type definitions for beats 0.0
+// Project: https://github.com/hughsk/beats/
+// Definitions by: Uri Shaked <https://github.com/urish>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = beats;
+
+type DetectFunction = (frequencies: Uint8Array | Float32Array | ReadonlyArray<number>, dt?: number) => Float32Array;
+
+declare function beats(bins: ReadonlyArray<beats.Bin>, minSeparation?: number): DetectFunction;
+
+declare namespace beats {
+    interface Bin {
```

Comment:
The `namespace` can be removed, the `interface Bin` can be just be exported outside of the namespace.

Reply:
Thank you for your feedback! I tried removing the namespace, but then the test fails with this error:

    Module '"/DefinitelyTyped/types/beats/index"' resolves to a non-module entity and cannot be imported using this construct.

on the following line:

    import * as beats from 'beats';

thoughts?

Reply:
You should use `import beats = require('beats');`, because ES6 module can not be used as function or class. https://stackoverflow.com/questions/39415661/what-does-resolves-to-a-non-module-entity-and-cannot-be-imported-using-this

Reply:
Thank you! I tried this, but then I couldn't figure out a way to import the `Bin` interface from the "beats" module, and thus I can't use the interface in my own code. Is there a way to make it work with this import syntax?

This is the current `index.ts` code:

```typescript
export = beats;

declare function beats(bins: ReadonlyArray<Bin>, minSeparation?: number):
    (frequencies: Uint8Array | Float32Array | ReadonlyArray<number>, dt?: number) => Float32Array;

interface Bin {
    lo: number;
    hi: number;
    threshold: number;
    decay: number;
}
```

and the relevant part of the test:

```typescript
import beats = require('beats');

const bins: beats.Bin[] = [
    { lo: 0, hi: 512, threshold: 0, decay: 0.005 },
    { lo: 512, hi: 1024, threshold: 0, decay: 0.005 },
];
```

It says ` Cannot find namespace 'beats' `

Reply:
Have you tried to use:
```ts
export interface Bin {
    lo: number;
    hi: number;
    threshold: number;
    decay: number;
}
```

Reply:
The function can be changed to:
```ts
export function beats(bins: ReadonlyArray<Bin>, minSeparation?: number):
    (frequencies: Uint8Array | Float32Array | ReadonlyArray<number>, dt?: number) => Float32Array;
```

Reply:
I have just tried this, but then I can't use the imported function.  This is my index file now:

```typescript
export function beats(bins: ReadonlyArray<Bin>, minSeparation?: number):
(frequencies: Uint8Array | Float32Array | ReadonlyArray<number>, dt?: number) => Float32Array;

export interface Bin {
    lo: number;
    hi: number;
    threshold: number;
    decay: number;
}
```

I get this error:

`Cannot invoke an expression whose type lacks a call signature. Type 'typeof "DefinitelyTyped/types/beats/index"' has no compatible call signatures`

on the following line in the test file:

`const detect = beats(bins, 1);`